/*
异常在子父类覆盖中的体现：
1.子类在覆盖父类时，如果父类的方法抛出异常，那么子类的覆盖方法，只能抛出父类的异常或者该异常的子类。


*/

class AException extends Exception
{

}

class BException extends AException
{
}

class CException extends Exception
{
}

/*
Exception
	|--AException
		|--BException
	|--CException
*/

class Fu
{
	void show()throws AException
	{

	}
}

class Test
{
	void function(Fu f)
	{
		try
		{
			f.show();
		}
		catch (AException e)
		{
		}
	}
}

class Zi extends Fu
{
	void show()throws CException
	{

	}
}

class ExceptionDemo7 
{
	public static void main(String[] args) 
	{
		Test t = new Test();
		t.function(new Zi());
	}
}
